code/controllers/subsystems/initialization/atlas.dm

```diff
@@ -174,3 +177,29 @@ var/datum/controller/subsystem/atlas/SSatlas
 /proc/commstation_name()
 	ASSERT(current_map)
 	return current_map.dock_name
+
+/datum/controller/subsystem/atlas/proc/find_away_mission()
+	var/list/away_maps = list()
+	var/datum/away_map/M
+	for (var/type in subtypesof(/datum/away_map))
+		M = new type
+		if (!M.path)
+			log_debug("SSaway: Map [M.name] ([M.type]) has no path set, discarding.")
+			qdel(M)
+			continue
+
+		away_maps[M] = M
+	var/datum/away_map/X = pick(away_maps)
+	var/datum/away_map/map_used = X.path
+	maploader.load_map(file("maps/away/[map_used]/[map_used].dmm"), no_changeturf = TRUE)
+	X.zloc = world.maxz
+	X.handle_random_gen()
```

Comment:
This should prob. be handled by the map finalization SS. Perhaps this entire chunk should go there?